samples/browser/src/main/java/org/mozilla/samples/browser/addons/InstalledAddonDetailsActivity.kt

```diff
@@ -93,13 +94,21 @@ class InstalledAddonDetailsActivity : AppCompatActivity() {
         }
     }
 
-    private fun bindSettings(addOn: Addon) {
+    private fun bindSettings(addon: Addon) {
         val view = findViewById<View>(R.id.settings)
-        view.isEnabled = addOn.installedState?.optionsPageUrl != null
+        val optionsPageUrl = addon.installedState?.optionsPageUrl
+        view.isEnabled = optionsPageUrl != null
         view.setOnClickListener {
-            val intent = Intent(this, AddonSettingsActivity::class.java)
-            intent.putExtra("add_on", addOn)
-            this.startActivity(intent)
+            if (addon.installedState?.openOptionsPageInTab == true) {
+                components.tabsUseCases.addTab(optionsPageUrl as String)
+                val intent = Intent(this, BrowserActivity::class.java)
+                intent.addFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP)
+                this.startActivity(intent)
+            } else {
+                val intent = Intent(this, AddonSettingsActivity::class.java)
+                intent.putExtra("add_on", addon)
```

Comment:
I think we should send `optionsPageUrl` and `title` here instead of sending add-on parcel (I think there was a discussion over slack about add-on parcel size being too large). Don't start this activity unless `addon.installedState?.openOptionsPageInTab` exists (maybe log it or show indication if null?). 

Otherwise, we may show a blank page in `AddonSettingsActivity` and this may get flagged as a bug :). What do you think?

Reply:
- We've fixed that add-on parcel size already. That was just because we had all translations in there as well.

> Don't start this activity unless addon.installedState?.openOptionsPageUrl exists (maybe log it or show indication if null?).
That can't happen as we only enable the button if it exists. That's why I made it nullable, or did I miss anything?